### PR TITLE
Provide special member functions for TriaAccessor

### PR DESCRIPTION
**Pull request by masterleinad**
Addresses https://github.com/dealii/dealii/issues/12117#issuecomment-846683808.

### CODE REVIEW

File: include/deal.II/grid/tria_accessor.h [hunk 1]
```diff
@@ -754,6 +764,17 @@ class TriaAccessor : public TriaAccessorBase<structdim, dim, spacedim>
   void
   operator=(const TriaAccessor &) = delete;
 
+  /**
+   * Move assignemt operator. Moving is allowed.
```
Comment (drwells) [inline include/deal.II/grid/tria_accessor.h:768]:
```suggestion
   * Move assignment operator. Moving is allowed.
```

File: include/deal.II/grid/tria_accessor.h [hunk 2]
```diff
@@ -754,6 +764,17 @@ class TriaAccessor : public TriaAccessorBase<structdim, dim, spacedim>
...
+   */
+  void
+  operator=(TriaAccessor &&) = default;
+
+  /**
+   * Defaulted destructor.
+   */
+  ~TriaAcessor() = default;
```
Comment (drwells) [inline include/deal.II/grid/tria_accessor.h:776]: Do we need this to avoid a warning?

Comment (masterleinad) [reply include/deal.II/grid/tria_accessor.h:776]: No, I am just following the rule of 5.

**Review by drwells — APPROVED** [review]
This is fine but there are a few typos (including the class name in the destructor).

**Comment by marcfehling** [comment]
```
/jenkins/workspace/dealii-tidy_PR-12294/include/deal.II/grid/tria_accessor.h:732:3: error: move constructors should be marked noexcept [performance-noexcept-move-constructor,-warnings-as-errors]
/jenkins/workspace/dealii-tidy_PR-12294/include/deal.II/grid/tria_accessor.h:772:3: error: move assignment operators should be marked noexcept [performance-noexcept-move-constructor,-warnings-as-errors]
```

**Review by marcfehling — APPROVED** [review]
Okay after `clang-tidy` is happy.

**Comment by marcfehling** [comment]
Is this supposed to be part of the `9.3.0` release?

**Pull request merged**